include/irr/asset/IDescriptor.h

```diff
@@ -0,0 +1,16 @@
+#ifndef __IRR_I_DESCRIPTOR_H_INCLUDED__
+#define __IRR_I_DESCRIPTOR_H_INCLUDED__
+
+namespace irr {
+namespace asset
+{
+
+class IDescriptor
+{
+public:
+    virtual ~IDescriptor() = default;
```

Comment:
you need a `virtual getType` and a descriptor type enum

Reply:
not a good idea because resources can be bound to more than one descriptor types. Especially buffers: ssbo, ubo, dynamic ssbo, dynamic ubo

Reply:
Now descriptor type is in IDescriptorSet::SWriteDescriptorSet::descriptorType and it's sufficient for this

Reply:
> not a good idea because resources can be bound to more than one descriptor types. Especially buffers: ssbo, ubo, dynamic ssbo, dynamic ubo

Ok

Reply:
resolved